### PR TITLE
[API] Management for admin users 

### PR DESCRIPTION
**Pull request by tuka217**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | no|
| Related tickets | - |
| License         | MIT |

### CODE REVIEW

File: docs/api/admin_users.rst [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
++----------+---------------------------------+
+| enabled  | Flag set if the user is enabled |
++----------+---------------------------------+
+
+If you request for more detailed data, you will receive an object with the following fields:
+
++-------------------+-------------------------------------------------------+
+| Field             | Description                                           |
++===================+=======================================================+
+| id                | Admin user is                                         |
```
Comment (Zales0123) [inline docs/api/admin_users.rst]: `Admin user's id`

File: docs/api/admin_users.rst [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+
+Admin User API response structure
+---------------------------------
+
+If you request an admin user via API, you will receive an object with the following fields:
+
++----------+---------------------------------+
+| Field    | Description                     |
++==========+=================================+
+| id       | Admin user is                   |
```
Comment (Zales0123) [inline docs/api/admin_users.rst]: `Admin user's id`

File: docs/api/admin_users.rst [hunk 3]
```diff
@@ -0,0 +1,584 @@
...
++----------+---------------------------------+
+| username | Admin user name                 |
```
Comment (Zales0123) [inline docs/api/admin_users.rst]: It should be `Admin user's...` here and everywhere below.

File: docs/api/admin_users.rst [hunk 4]
```diff
@@ -0,0 +1,584 @@
...
++-------------------+-------------------------------------------------------+
+| email             | Admin user email                                      |
++-------------------+-------------------------------------------------------+
+| enabled           | Flag set if the user is enabled                       |
++-------------------+-------------------------------------------------------+
+| usernameCanonical | Username of the admin user in canonical form          |
++-------------------+-------------------------------------------------------+
+| emailCanonical    | Email of the admin user in canonical form             |
++-------------------+-------------------------------------------------------+
+| roles             | Roles of the admin users                              |
```
Comment (Zales0123) [inline docs/api/admin_users.rst]: `Roles of the admin user` or `Admin user's roles`

File: docs/api/admin_users.rst [hunk 5]
```diff
@@ -0,0 +1,584 @@
...
+| plainPassword | request        | Admin user password                                   |
++---------------+----------------+-------------------------------------------------------+
+| localeCode    | request        | Code of the language, which is used by the admin user |
++---------------+----------------+-------------------------------------------------------+
+
+
+Example
+^^^^^^^
+
+To create an new admin user use the below method:
```
Comment (Zales0123) [inline docs/api/admin_users.rst]: `a new`

File: docs/api/admin_users.rst [hunk 6]
```diff
@@ -0,0 +1,584 @@
...
+
+.. code-block:: text
+
+    STATUS: 201 CREATED
+
+.. code-block:: json
+
+    {
+        "id": 7,
+        "username": "Barlog",
```
Comment (Zales0123) [inline docs/api/admin_users.rst]: `Barlog` -> `Balrog`, but I respect this example 😄

File: docs/api/admin_users.rst [hunk 7]
```diff
@@ -0,0 +1,584 @@
...
+            "ROLE_ADMINISTRATION_ACCESS"
+        ],
+        "email": "teamEvil@middleearth.com",
+        "emailCanonical": "teamevil@middleearth.com",
+        "enabled": false
+    }
+
+.. warning::
+
+    If you try to create an admin user without username, email, password or locale's code , you will receive a ``400 Bad Request`` error, that will contain validation errors.
```
Comment (Zales0123) [inline docs/api/admin_users.rst]: Extra space before comma.

File: docs/api/admin_users.rst [hunk 8]
```diff
@@ -0,0 +1,584 @@
...
+.. code-block:: bash
+
+    $ curl http://demo.sylius.org/api/v1/users/ \
+        -H "Authorization: Bearer SampleToken" \
+        -H "Content-Type: application/json" \
+        -X POST \
+        --data '
+            {
+                "firstName": "Barlog",
+                "lastName": "Barlog!",
```
Comment (Zales0123) [inline docs/api/admin_users.rst]: I would vote for the `of Morgoth` as the last name, as this creature is widely known as **Balrog of Morgoth** ;)

File: docs/api/admin_users.rst [hunk 9]
```diff
@@ -0,0 +1,584 @@
...
+Exemplary Response
+^^^^^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+    STATUS: 204 No Content
+
+.. warning::
+
+    If you try to delete the admin user which is currently logged, you will receive a ``422 Unprocessable Entity`` error.
```
Comment (Zales0123) [inline docs/api/admin_users.rst]: `currently logged` -> `currently logged in`?

File: tests/Controller/AdminUserApiTest.php [hunk 10]
```diff
@@ -0,0 +1,355 @@
...
+        $this->client->request('GET', $this->getAdminUserUrl($users['admin1']), [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'admin_user/show_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_denies_full_admin_user_update_for_not_authenticated_users()
```
Comment (Zales0123) [inline tests/Controller/AdminUserApiTest.php]: `it_denies_full_admin_user_update` -> `it_denies_admin_user_full_update`?

File: tests/Controller/AdminUserApiTest.php [hunk 11]
```diff
@@ -0,0 +1,355 @@
...
+
+        $response = $this->client->getResponse();
+
+        $this->assertResponse($response, 'authentication/access_denied_response', Response::HTTP_UNAUTHORIZED);
+    }
+
+    /**
+     * @test
+     */
+    public function it_returns_not_found_response_when_requesting_full_update_of_a_admin_user_which_does_not_exist()
```
Comment (Zales0123) [inline tests/Controller/AdminUserApiTest.php]: `an_admin_user` here and in some other places in this file.

**Comment by pjedrzejewski** [comment]
@tuka217 Please apply fixes according to @Zales0123's review and it is good to go.

File: docs/api/admin_users.rst (hunk 8 above)
Comment (lchrusciel) [inline docs/api/admin_users.rst]: This change is a blocker for me. We should change it ASAP.

**Comment by pjedrzejewski** [comment]
Thank you Ania! :)

**Pull request merged**